### ISSUE TITLE
Add a flag logo for each event

### ISSUE THREAD

**Issue by Binnette** [thomersch/openstreetmap-calendar#46]
Hi, like in the wiki: 
![image](https://user-images.githubusercontent.com/6597721/95757024-e02bbf80-0ca6-11eb-9a7f-34730a927d48.png)

Is it possible to add a country flag for each event? This is very useful, i can see instantly in which country the event take place.

EDIT: In fact, i don't really care about the country for an online event. Maybe you should consider adding a flag that indicate which language will be spoken during the event? What do you think?

Knowing the spoken language is highly important for online event. I don't want to attend an event in a language that I can non understand.

Have a nice day.

**Comment by thomersch** [comment]
Thanks for reporting. Countries, nation states and languages are a highly delicate topic and I don't want to do half-baked, possibly offensive changes to software:

Flags have a variety of problems, which mean that I will probably not integrate them. If we start to have hundreds of upcoming events, a language filter might start to be useful, but [flags do not represent languages](http://www.flagsarenotlanguages.com/blog/why-flags-do-not-represent-language/) and all human interface guidelines I've come across explicitly forbid to do so.

Registering a language for an event is also highly problematic, since this is not clearly defined what is spoken at a place. Also, in some countries people are willing to adapt to speakers of other languages.

I understand your use case here, but I think that most people wouldn't use such a feature after all.

**Comment by Binnette** [comment]
> I understand your use case here, but I think that most people wouldn't use such a feature after all.

At least me, i will 😅
Yes, cause i'm interested in joining events in French and English 😉 

So i will need to read every line, one by one 😅

Have a nice day.

**Comment by dieterdreist** [comment]
clearly flags are not good to represent languages, but this is about events. Would you not want to know where the events are, more or less, without reading the whole wall of text?  I would.

**Comment by danieldegroot2** [comment]
@dieterdreist 
- Country flags are still being displayed by [WeeklyOSM](https://weeklyosm.eu/)
- On the [wiki](https://wiki.openstreetmap.org/wiki/Main_Page) you can Ctrl + F to search. The "City", "Country" and "Type" icons are restricted to the disused [Calendar/events](https://wiki.openstreetmap.org/wiki/Template:Calendar/event) sub-template. Someone could do something similar as you can see from WeeklyOSM, but likely not on the wiki, to avoid clutter and long loading times.
- A map is available on the [OpenStreetMap Calendar Widget](https://jbelien.github.io/openstreetmap-calendar-widget/) which you can also deploy yourself ([GitHub](https://github.com/jbelien/openstreetmap-calendar-widget))
- On [osmcal.org](https://osmcal.org/) (and own deployed widgets, such as on wiki pages) you can get a feed/filter by country (see also #67 for cities).

Regarding your idea; some users may
- struggle to quickly identify if it’s displaying a location or a language.
- consider city/region/area specific flags as usable for them, but they may look like clutter to others.
- have difficulty finding them when scrolling, especially for smaller areas; here filters are better imo. I would leave custom layout to third parties.
- may not be aware certain flags are for rather nearby areas.

WeeklyOSM has a compact, custom layout with a clear "Country" label for the flags column(and seperate "location" column), whereas the osmcal.org website tries to provide a clean, basic interface.

It may be possible to add flags to the country filter list. See #25 - this would have to be resolved.
Disputed areas may or may not be an issue (page on [Wikipedia](https://en.wikipedia.org/wiki/List_of_territorial_disputes), [OSM Wiki](https://wiki.openstreetmap.org/wiki/Disputed_territories#List_of_cases_and_their_representation_in_OSM)).

**Comment by dieterdreist** [comment]
thank you for taking the time to reply, from my point of view (user of the OpenStreetMap wiki and organizer of a recurring event) your project has changed usability in the backend (instead of copying the former event line of code, adjusting the date and repasting it, we now have a convenient process) for a wall of text in the presentation view rather than a wiki table with flags and horizontal separations.

The map is neither linked, nor does it seem enabling to get the information quick, (you‘d have to zoom and pan around a lot), this is to illustrate what I see:

![FEE51169-CF41-45E8-ABE7-A770AE901ADE](https://user-images.githubusercontent.com/581288/164025857-98547bab-719b-473b-ab5b-6a001c95ab1a.png)


this is how the „calendar“ now looks in the wiki:

![37D3212B-90E4-4BFB-AE23-789507A4D4E5](https://user-images.githubusercontent.com/581288/164026000-2852d331-4675-47e0-a68c-1f6b569da907.png)

**Comment by danieldegroot2** [comment]
@dieterdreist 
For the wiki, could you let other wiki users know you'd like to reimplement icons via this wiki page, please?
https://wiki.openstreetmap.org/wiki/Talk:Wiki

**Comment by danieldegroot2** [comment]
@dieterdreist 
I understand your issue and I'd like your feature suggestion to work for a large number of people where possible, as osmcal.org itself is a small project and general-purpose viewer from which others can make their own feeds and versions.
I'm not one of the osmcal.org \ Widgets developers or Wiki \ Weekly admins and I can't speak for them. Don't worry, this misinterpretation can happen. (I changed the first bit of text I wrote regarding the wiki to be more coherent, btw.)
